mongoOperator/helpers/RestoreHelper.py

```diff
@@ -0,0 +1,174 @@
+        )
+
+    @staticmethod
+    def _lastBackupFile(credentials: dict, bucket_name: str, key: str) -> str:
+        """
+        Gets the name of the last backup file in the bucket.
+        :param credentials: The Google cloud storage service credentials retrieved from the Kubernetes secret.
+        :param bucket_name: The name of the bucket.
+        :param key: The prefix of tha backups
+        :return: The location of the last backup file.
+        """
+        credentials = ServiceCredentials.from_service_account_info(credentials)
+        gcs_client = StorageClient(credentials.project_id, credentials)
+        bucket = gcs_client.get_bucket(bucket_name)
+        blobs = bucket.list_blobs(prefix=key)
+
+        last_blob = None
+        for blob in blobs:
+            logging.info("Found backup file '%s' in bucket '%s'", blob.name, bucket_name)
+            if last_blob is None or blob.time_created > last_blob.time_created:
+                last_blob = blob
+
+        logging.info("Returning backup file %s", last_blob.name.replace(key, ""))
+        return last_blob.name.replace(key, "") if last_blob else None
+
+    def restoreIfNeeded(self, cluster_object: V1MongoClusterConfiguration) -> bool:
+        """
+        Checks whether a restore is requested for the cluster, looking up the restore file if
+        necessary.
+        :param cluster_object: The cluster object from the YAML file.
+        :return: Whether a restore was executed or not.
+        """
+        if cluster_object.spec.backups.gcs.restore_from is not None:
+            backup_file = cluster_object.spec.backups.gcs.restore_from
+            if backup_file == 'latest':
+                backup_file = self.getLastBackup(cluster_object)
+
+            logging.info("Attempting to restore file %s to cluster %s @ ns/%s.", backup_file,
+                         cluster_object.metadata.name, cluster_object.metadata.namespace)
+
+            self.restore(cluster_object, backup_file)
+            return True
+
+        return False
+
+    def restore(self, cluster_object: V1MongoClusterConfiguration, backup_file: str) -> bool:
+        """
+        Attempts to restore the latest backup in the specified location to the given cluster.
+        Creates a new backup for the given cluster saving it in the cloud storage.
+        :param cluster_object: The cluster object from the YAML file.
+        :param backup_file: The filename of the backup we want to restore.
+        """
+        hosts = MongoResources.getConnectionSeeds(cluster_object)
+
+        logging.info("Restoring backup file %s to cluster %s @ ns/%s.", backup_file,
+                     cluster_object.metadata.name, cluster_object.metadata.namespace)
+
+        # Download the backup file from the bucket
+        downloaded_file = self._downloadBackup(cluster_object, backup_file)
+
+        for _ in range(self.RESTORE_RETRIES):
+            # Wait for the replicaset to become ready
```

Comment:
We can put this comment line above the for loop and remove the spaces between the for statement and the try statement.